source/standalone/tutorials/04_sensors/benchmark_cameras.py

```diff
@@ -0,0 +1,668 @@
+
+def run_simulator(
+    sim: sim_utils.SimulationContext,
+    scene_entities: dict,
+    warm_start_length: int = 10,
+    experiment_length: int = 100,
+    tiled_camera_replicators: list[str] | None = None,
+    standard_camera_replicators: list[str] | None = None,
+    ray_caster_camera_replicators: list[str] | None = None,
+    depth_predicate: Callable = lambda x: "to" in x or x == "depth",
+    perspective_depth_predicate: Callable = lambda x: x == "depth" or x == "depth_to_camera",
+    convert_depth_to_camera_to_image_plane: bool = True,
+    visualize: bool = False,
+) -> dict:
+    """Run the simulator with all cameras, and return timing analytics. Visualize if desired."""
+
+    if tiled_camera_replicators is None:
+        tiled_camera_replicators = ["rgb"]
+    if standard_camera_replicators is None:
+        standard_camera_replicators = ["rgb"]
+    if ray_caster_camera_replicators is None:
+        ray_caster_camera_replicators = ["rgb"]
+    # Extract entities for simplified notation
+    num_tiled_cameras = 0
+    tiled_camera = None
+    num_standard_cameras = 0
+    standard_camera = None
+    num_ray_caster_cameras = 0
+    ray_caster_camera = None
+
+    if "tiled_camera" in scene_entities:
+        tiled_camera: TiledCamera = scene_entities["tiled_camera"]
+        num_tiled_cameras = tiled_camera.data.intrinsic_matrices.size(0)
+    if "standard_camera" in scene_entities:
+        standard_camera: Camera = scene_entities["standard_camera"]
+        num_standard_cameras = standard_camera.data.intrinsic_matrices.size(0)
+    if "ray_caster_camera" in scene_entities:
+        ray_caster_camera: RayCasterCamera = scene_entities["ray_caster_camera"]
+        num_ray_caster_cameras = ray_caster_camera.data.intrinsic_matrices.size(0)
+
+    camera_counts = [num_tiled_cameras, num_standard_cameras, num_ray_caster_cameras]
+    cameras = [tiled_camera, standard_camera, ray_caster_camera]
+    all_replicators = [tiled_camera_replicators, standard_camera_replicators, ray_caster_camera_replicators]
+    labels = ["tiled", "standard", "ray_caster"]
+
+    for num, camera in zip(camera_counts, cameras):
+        if num > 0:
+            positions = torch.tensor([[2.5, 2.5, 2.5]], device=sim.device).repeat(num, 1)
+            targets = torch.tensor([[0.0, 0.0, 0.0]], device=sim.device).repeat(num, 1)
+            camera.set_world_poses_from_view(positions, targets)
+
+    # Initialize timing variables
+    timestep = 0
+    total_time = 0.0
+    valid_timesteps = 0
+    sim_step_time = 0.0
+    vision_processing_time = 0.0
+
+    while simulation_app.is_running() and timestep < experiment_length:
+        print(f"On timestep {timestep} of {experiment_length}, with warm start of {warm_start_length}")
+
+        # Measure simulation step time
+        sim_start_time = time.time()
+
+        sim.step()
+        sim_end_time = time.time()
+        sim_step_time += sim_end_time - sim_start_time
+
+        if timestep > warm_start_length:
```

Comment:
Why is this warm start length done?

Reply:
To deal with when the first couple of frames can be blank: 

https://isaac-sim.github.io/IsaacLab/source/refs/issues.html#blank-initial-frames-from-the-camera